Problem build with Matlab2018b on the mac

**Describe the bug**
Problem build with Matlab2018b on the mac

**What is the expected behavior?**
 What is the motivation / use case for changing the behavior?

**To Reproduce**
Steps to reproduce the behavior:

```
$ export PATH="/Applications/MATLAB_R2018b.app/bin/:$PATH"
$ cmake -DBUILD_MATLAB_INTERFACE=ON -DCMAKE_INSTALL_PREFIX="/Users/$USER/local/helics-master" ..
-- setting up for AppleClang on Darwin-16.7.0
-- setting helics C++ standard build option to "-std=c++14"
-- Found Boost: /Users/jsimpson/anaconda3/envs/go_solar36/include (found suitable version "1.68.0", minimum required is "1.58")  
-- Found ZeroMQ library: /Users/jsimpson/anaconda3/envs/go_solar36/lib/libzmq.dylib
-- Found ZeroMQ headers: /Users/jsimpson/anaconda3/envs/go_solar36/include
-- Version: 6.0.0
-- Build type: 
-- CXX_STANDARD: 14
-- JsonCpp Version: 1.9.2
-- Found Boost: /Users/jsimpson/anaconda3/envs/go_solar36/include (found suitable version "1.68.0", minimum required is "1.61")  
-- Boost dir= /Users/jsimpson/anaconda3/envs/go_solar36/include
-- Found Matlab: /Applications/MATLAB_R2018b.app/extern/include (found version "9.5") found components:  MAIN_PROGRAM 
-- Building MATLAB
-- Extra /Users/jsimpson/git/GO-Solar/HELICS/interfaces/matlab/extra/helicsInputSetDefault.m;/Users/jsimpson/git/GO-Solar/HELICS/interfaces/matlab/extra/helicsPublicationPublish.m
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/jsimpson/git/GO-Solar/HELICS/build-osx
(go_solar36) jsimpson-31689s:build-osx jsimpson$ make -j8
[  3%] Built target jsoncpp_lib
[ 10%] Built target gmlc_utilities
[ 13%] Built target units-static
[ 14%] Built target helics-config
[ 16%] Built target fmt
Scanning dependencies of target helics_common
[ 18%] Building CXX object src/helics/common/CMakeFiles/helics_common.dir/TomlProcessingFunctions.cpp.o
[ 18%] Building CXX object src/helics/common/CMakeFiles/helics_common.dir/JsonProcessingFunctions.cpp.o
[ 18%] Building CXX object src/helics/common/CMakeFiles/helics_common.dir/JsonBuilder.cpp.o
[ 19%] Linking CXX static library libhelics_common.a
[ 23%] Built target helics_common
Scanning dependencies of target helics_core
[ 24%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/CoreFactory.cpp.o
[ 24%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/BrokerFactory.cpp.o
[ 25%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/BrokerBase.cpp.o
[ 26%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/CommonCore.cpp.o
[ 27%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/FederateState.cpp.o
[ 28%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/NamedInputInfo.cpp.o
[ 29%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/InterfaceInfo.cpp.o
[ 30%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/FilterInfo.cpp.o
[ 30%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/EndpointInfo.cpp.o
[ 31%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/ActionMessage.cpp.o
[ 32%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/CoreBroker.cpp.o
[ 33%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/TimeCoordinator.cpp.o
[ 33%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/NetworkCommsInterface.cpp.o
[ 34%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/ForwardingTimeCoordinator.cpp.o
[ 35%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/TimeDependencies.cpp.o
[ 36%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/CommsInterface.cpp.o
[ 36%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/CommsBroker.cpp.o
[ 37%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/NetworkBrokerData.cpp.o
[ 38%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/HandleManager.cpp.o
[ 39%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/FilterCoordinator.cpp.o
[ 40%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/TimeoutMonitor.cpp.o
[ 41%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/coreTypeOperations.cpp.o
[ 41%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/MessageTimer.cpp.o
[ 42%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/inproc/InprocBroker.cpp.o
[ 43%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/inproc/InprocCore.cpp.o
[ 44%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/inproc/InprocComms.cpp.o
[ 44%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/udp/UdpCore.cpp.o
[ 45%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/udp/UdpBroker.cpp.o
[ 46%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/udp/UdpComms.cpp.o
[ 47%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/ipc/IpcCore.cpp.o
[ 47%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/ipc/IpcBroker.cpp.o
[ 48%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/ipc/IpcComms.cpp.o
[ 49%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/ipc/IpcQueueHelper.cpp.o
[ 50%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/tcp/TcpCore.cpp.o
[ 50%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/tcp/TcpBroker.cpp.o
[ 51%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/tcp/TcpComms.cpp.o
[ 52%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/tcp/TcpCommsSS.cpp.o
[ 52%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/tcp/TcpCommsCommon.cpp.o
[ 53%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/zmq/ZmqCore.cpp.o
[ 54%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/zmq/ZmqBroker.cpp.o
[ 55%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/zmq/ZmqComms.cpp.o
[ 55%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/zmq/ZmqCommsSS.cpp.o
[ 56%] Building CXX object src/helics/core/CMakeFiles/helics_core.dir/zmq/ZmqRequestSets.cpp.o
[ 57%] Linking CXX static library libhelics_core.a
[ 60%] Built target helics_core
Scanning dependencies of target helics_application_api
[ 61%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/CombinationFederate.cpp.o
[ 62%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/Federate.cpp.o
[ 62%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/MessageFederateManager.cpp.o
[ 63%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/MessageFederate.cpp.o
[ 64%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/MessageOperators.cpp.o
[ 65%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/ValueFederate.cpp.o
[ 66%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/ValueConverter.cpp.o
[ 66%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/ValueFederateManager.cpp.o
[ 67%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/helicsPrimaryTypes.cpp.o
[ 68%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/Publications.cpp.o
[ 69%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/Filters.cpp.o
[ 69%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/FilterOperations.cpp.o
[ 70%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/FilterFederateManager.cpp.o
[ 71%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/Endpoints.cpp.o
[ 72%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/helicsTypes.cpp.o
[ 72%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/queryFunctions.cpp.o
[ 73%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/FederateInfo.cpp.o
[ 74%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/Inputs.cpp.o
[ 75%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/BrokerApp.cpp.o
[ 75%] Building CXX object src/helics/application_api/CMakeFiles/helics_application_api.dir/CoreApp.cpp.o
[ 76%] Linking CXX static library libhelics_application_api.a
[ 76%] Built target helics_application_api
Scanning dependencies of target helicsSharedLib
Scanning dependencies of target helics_apps
[ 77%] Building CXX object src/helics/shared_api_library/CMakeFiles/helicsSharedLib.dir/ValueFederateExport.cpp.o
[ 77%] Building CXX object src/helics/shared_api_library/CMakeFiles/helicsSharedLib.dir/MessageFiltersExport.cpp.o
[ 78%] Building CXX object src/helics/shared_api_library/CMakeFiles/helicsSharedLib.dir/MessageFederateExport.cpp.o
[ 78%] Building CXX object src/helics/shared_api_library/CMakeFiles/helicsSharedLib.dir/helicsExport.cpp.o
[ 79%] Building CXX object src/helics/shared_api_library/CMakeFiles/helicsSharedLib.dir/helicsCallbacks.cpp.o
[ 80%] Building CXX object src/helics/apps/CMakeFiles/helics_apps.dir/Recorder.cpp.o
[ 81%] Building CXX object src/helics/shared_api_library/CMakeFiles/helicsSharedLib.dir/FederateExport.cpp.o
[ 82%] Building CXX object src/helics/apps/CMakeFiles/helics_apps.dir/Player.cpp.o
[ 82%] Building CXX object src/helics/apps/CMakeFiles/helics_apps.dir/PrecHelper.cpp.o
[ 83%] Building CXX object src/helics/apps/CMakeFiles/helics_apps.dir/SignalGenerators.cpp.o
[ 84%] Building CXX object src/helics/apps/CMakeFiles/helics_apps.dir/Echo.cpp.o
[ 85%] Building CXX object src/helics/apps/CMakeFiles/helics_apps.dir/Source.cpp.o
[ 85%] Building CXX object src/helics/apps/CMakeFiles/helics_apps.dir/Tracer.cpp.o
[ 86%] Building CXX object src/helics/apps/CMakeFiles/helics_apps.dir/helicsApp.cpp.o
[ 87%] Linking CXX shared library libhelicsSharedLib.dylib
[ 88%] Building CXX object src/helics/apps/CMakeFiles/helics_apps.dir/Clone.cpp.o
[ 88%] Built target helicsSharedLib
Scanning dependencies of target helicsMEX
[ 89%] Generating helicsMEX.cpp
[ 90%] Generating mkhelicsMEXFile.m
[ 91%] Generating helicsMEX.mexmaci64

                                                                                  < M A T L A B (R) >
                                                                        Copyright 1984-2018 The MathWorks, Inc.
                                                                    R2018b Update 5 (9.5.0.1178774) 64-bit (maci64)
                                                                                    August 6, 2019

Warning: Function echo has the same name as a MATLAB builtin. We suggest you rename the function to avoid a potential name conflict. 
 
For online documentation, see https://www.mathworks.com/support
For product information, visit www.mathworks.com.
 

	Trial License -- for use to evaluate programs for possible purchase as an end-user only.

Building with 'Xcode Clang++'.
[ 92%] Building CXX object src/helics/apps/CMakeFiles/helics_apps.dir/MultiBroker.cpp.o
[ 92%] Building CXX object src/helics/apps/CMakeFiles/helics_apps.dir/BrokerServer.cpp.o
Error using mex
/Users/jsimpson/git/GO-Solar/HELICS/build-osx/interfaces/matlab/helicsMEX.cpp:2342:14: warning: implicit conversion of NULL constant to 'int' [-Wnull-conversion]
      return NULL;
      ~~~~~~ ^~~~
             0
/Users/jsimpson/git/GO-Solar/HELICS/build-osx/interfaces/matlab/helicsMEX.cpp:2574:14: warning: implicit conversion of NULL constant to 'int' [-Wnull-conversion]
      return NULL;
      ~~~~~~ ^~~~
             0
/Users/jsimpson/git/GO-Solar/HELICS/build-osx/interfaces/matlab/helicsMEX.cpp:4072:14: warning: implicit conversion of NULL constant to 'int' [-Wnull-conversion]
      return NULL;
      ~~~~~~ ^~~~
             0
/Users/jsimpson/git/GO-Solar/HELICS/build-osx/interfaces/matlab/helicsMEX.cpp:8694:14: warning: implicit conversion of NULL constant to 'int' [-Wnull-conversion]
      return NULL;
      ~~~~~~ ^~~~
             0
/Users/jsimpson/git/GO-Solar/HELICS/build-osx/interfaces/matlab/helicsMEX.cpp:8928:10: error: use of undeclared identifier 'helicsSubscriptionGetValueSize'; did you mean
'helicsSubscriptionGetKey'?
    arg3=helicsSubscriptionGetValueSize(arg1)+2;
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
         helicsSubscriptionGetKey
/Users/jsimpson/git/GO-Solar/HELICS/src/helics/shared_api_library/ValueFederate.h:615:31: note: 'helicsSubscriptionGetKey' declared here
    HELICS_EXPORT const char *helicsSubscriptionGetKey (helics_input ipt);
                              ^
/Users/jsimpson/git/GO-Solar/HELICS/build-osx/interfaces/matlab/helicsMEX.cpp:8928:46: error: assigning to 'int' from incompatible type 'const char *'
    arg3=helicsSubscriptionGetValueSize(arg1)+2;
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
/Users/jsimpson/git/GO-Solar/HELICS/build-osx/interfaces/matlab/helicsMEX.cpp:8935:30: error: no matching function for call to 'SWIG_FromCharPtrAndSize'
    if (--resc>=0) *resv++ = SWIG_FromCharPtrAndSize(arg2,*arg4);
                             ^~~~~~~~~~~~~~~~~~~~~~~
/Users/jsimpson/git/GO-Solar/HELICS/build-osx/interfaces/matlab/helicsMEX.cpp:1366:1: note: candidate function not viable: cannot convert argument of incomplete type 'void *' to 'const
char *' for 1st argument
SWIG_FromCharPtrAndSize(const char* carray, size_t size)
^
/Users/jsimpson/git/GO-Solar/HELICS/build-osx/interfaces/matlab/helicsMEX.cpp:9867:14: warning: implicit conversion of NULL constant to 'int' [-Wnull-conversion]
      return NULL;
      ~~~~~~ ^~~~
             0
/Users/jsimpson/git/GO-Solar/HELICS/build-osx/interfaces/matlab/helicsMEX.cpp:11889:30: error: no matching function for call to 'SWIG_FromCharPtrAndSize'
    if (--resc>=0) *resv++ = SWIG_FromCharPtrAndSize(arg2,*arg4);
                             ^~~~~~~~~~~~~~~~~~~~~~~
/Users/jsimpson/git/GO-Solar/HELICS/build-osx/interfaces/matlab/helicsMEX.cpp:1366:1: note: candidate function not viable: cannot convert argument of incomplete type 'void *' to 'const
char *' for 1st argument
SWIG_FromCharPtrAndSize(const char* carray, size_t size)
^
5 warnings and 4 errors generated.


Error in mkhelicsMEXFile (line 1)
mex('-I/Users/jsimpson/git/GO-Solar/HELICS/src/helics/shared_api_library/','-I/Users/jsimpson/git/GO-Solar/HELICS/build-osx/src/helics/shared_api_library/',
'-L/Users/jsimpson/git/GO-Solar/HELICS/build-osx/src/helics/shared_api_library','-lhelicsSharedLib',
'-largeArrayDims','/Users/jsimpson/git/GO-Solar/HELICS/build-osx/interfaces/matlab/helicsMEX.cpp')

Error in run (line 91)
evalin('caller', strcat(script, ';'));
 
>> [ 93%] Linking CXX static library libhelics_apps.a
[ 93%] Built target helics_apps
Scanning dependencies of target helics_broker_server
Scanning dependencies of target helics_broker
Scanning dependencies of target helics_recorder
Scanning dependencies of target helics_app
[ 93%] Building CXX object src/helics/apps/CMakeFiles/helics_broker_server.dir/helics-broker-server.cpp.o
[ 94%] Building CXX object src/helics/apps/CMakeFiles/helics_broker.dir/helics-broker.cpp.o
[ 95%] Building CXX object src/helics/apps/CMakeFiles/helics_recorder.dir/recorderMain.cpp.o
[ 96%] Building CXX object src/helics/apps/CMakeFiles/helics_app.dir/appMain.cpp.o
Scanning dependencies of target helics_player
[ 96%] Building CXX object src/helics/apps/CMakeFiles/helics_player.dir/playerMain.cpp.o
[ 97%] Linking CXX executable helics_recorder
[ 98%] Linking CXX executable helics_player
[ 98%] Built target helics_player
[ 98%] Built target helics_recorder
[ 99%] Linking CXX executable helics_broker_server
[100%] Linking CXX executable helics_broker
[100%] Built target helics_broker_server
[100%] Linking CXX executable helics_app
[100%] Built target helics_broker
[100%] Built target helics_app

>> 


```


**Environment (please complete the following information):**
Mac 10.12.6, Matlab 1018b, 
Xcode 9.2 per **https://stackoverflow.com/questions/51405776/how-can-i-get-xcode-working-on-my-mac-sierra-10-12-6**
Conda build env, I try not to use brew for the compilers. 
```
$ conda list
# packages in environment at /Users/jsimpson/anaconda3/envs/go_solar36:
#
# Name                    Version                   Build  Channel
appnope                   0.1.0                 py36_1000    conda-forge
attrs                     19.3.0                     py_0    conda-forge
autoconf                  2.69            pl526h1de35cc_9    conda-forge
automake                  1.16.1                  pl526_0    anaconda
backcall                  0.1.0                      py_0    conda-forge
bleach                    3.1.0                      py_0    conda-forge
blosc                     1.17.0               h6de7cb9_1    conda-forge
boost                     1.67.0                   py36_4    anaconda
boost-cpp                 1.68.0            h6f8c590_1000    conda-forge
bzip2                     1.0.8                h01d97ff_1    conda-forge
ca-certificates           2019.9.11            hecc5488_0    conda-forge
certifi                   2019.9.11                py36_0    conda-forge
cffi                      1.12.3                   pypi_0    pypi
cmake                     3.15.5               h6c18c4b_0    conda-forge
curl                      7.65.3               h22ea746_0    conda-forge
cycler                    0.10.0                   pypi_0    pypi
dbus                      1.13.6               h2f22bb5_0    conda-forge
decorator                 4.4.1                      py_0    conda-forge
defusedxml                0.6.0                      py_0    conda-forge
dss-python                0.10.1                   pypi_0    pypi
entrypoints               0.3                   py36_1000    conda-forge
expat                     2.2.5             h4a8c4bd_1004    conda-forge
future                    0.17.1                   pypi_0    pypi
gettext                   0.19.8.1          h46ab8bc_1002    conda-forge
glib                      2.58.3            h9d45998_1002    conda-forge
hdf5                      1.10.5          nompi_h3e39495_1104    conda-forge
helics                    2.0.0            py36hea25215_1    conda-forge
icu                       58.2              h0a44026_1000    conda-forge
importlib_metadata        0.23                     py36_0    conda-forge
ipykernel                 5.1.3            py36h5ca1d4c_0    conda-forge
ipython                   7.9.0            py36h5ca1d4c_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.5.1                      py_0    conda-forge
jedi                      0.15.1                   py36_0    conda-forge
jinja2                    2.10.3                     py_0    conda-forge
jpeg                      9c                h1de35cc_1001    conda-forge
jsonschema                3.1.1                    py36_0    conda-forge
jupyter                   1.0.0                      py_2    conda-forge
jupyter_client            5.3.3                    py36_1    conda-forge
jupyter_console           6.0.0                      py_0    conda-forge
jupyter_core              4.5.0                      py_0    conda-forge
kiwisolver                1.1.0                    pypi_0    pypi
krb5                      1.16.3            hcfa6398_1001    conda-forge
libblas                   3.8.0               14_openblas    conda-forge
libboost                  1.67.0               hebc422b_4    anaconda
libcblas                  3.8.0               14_openblas    conda-forge
libcurl                   7.65.3               h16faf7d_0    conda-forge
libcxx                    9.0.0                h89e68fa_1    conda-forge
libedit                   3.1.20181209         hb402a30_0  
libffi                    3.2.1             h6de7cb9_1006    conda-forge
libgfortran               4.0.0                         2    conda-forge
libiconv                  1.15              h01d97ff_1005    conda-forge
liblapack                 3.8.0               14_openblas    conda-forge
libopenblas               0.3.7                h4bb4525_3    conda-forge
libpng                    1.6.37               h2573ce8_0    conda-forge
libsodium                 1.0.17               h01d97ff_0    conda-forge
libssh2                   1.8.2                hcdc9a53_2    conda-forge
libuv                     1.33.1               h01d97ff_0    conda-forge
llvm-openmp               9.0.0                h40edb58_0    conda-forge
m4                        1.4.18            h1de35cc_1001    conda-forge
markupsafe                1.1.1            py36h0b31af3_0    conda-forge
matlabengineforpython     R2018a                   pypi_0    pypi
matplotlib                3.1.1                    pypi_0    pypi
mistune                   0.8.4           py36h0b31af3_1000    conda-forge
mock                      3.0.5                    py36_0    conda-forge
more-itertools            7.2.0                      py_0    conda-forge
nbconvert                 5.6.1                    py36_0    conda-forge
nbformat                  4.4.0                      py_1    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
networkx                  2.3                      pypi_0    pypi
notebook                  6.0.1                    py36_0    conda-forge
numexpr                   2.7.0            py36h4f17bb1_0    conda-forge
numpy                     1.17.3           py36hde6bac1_0    conda-forge
opendssdirect-py          0.3.7                    pypi_0    pypi
openssl                   1.1.1d               h0b31af3_0    conda-forge
pandas                    0.25.1                   pypi_0    pypi
pandoc                    2.7.3                         0    conda-forge
pandocfilters             1.4.2                      py_1    conda-forge
parso                     0.5.1                      py_0    conda-forge
pcre                      8.43                 h4a8c4bd_0    conda-forge
perl                      5.26.2            haec8ef5_1006    conda-forge
pexpect                   4.7.0                    py36_0    conda-forge
pickleshare               0.7.5                 py36_1000    conda-forge
pip                       19.2.3                   py36_0    conda-forge
prometheus_client         0.7.1                      py_0    conda-forge
prompt_toolkit            2.0.10                     py_0    conda-forge
ptyprocess                0.6.0                   py_1001    conda-forge
py-boost                  1.67.0           py36h6440ff4_4    anaconda
pycparser                 2.19                     pypi_0    pypi
pygments                  2.4.2                      py_0    conda-forge
pyparsing                 2.4.2                    pypi_0    pypi
pypower                   5.1.4                    pypi_0    pypi
pyqt                      5.9.2            py36h2a560b1_4    conda-forge
pyrsistent                0.15.5           py36h0b31af3_0    conda-forge
pytables                  3.6.1            py36h6f8395a_0    conda-forge
python                    3.6.9                h359304d_0  
python-dateutil           2.8.1                      py_0    conda-forge
pytz                      2019.2                   pypi_0    pypi
pyzmq                     18.1.0           py36h4bf09a9_0    conda-forge
qt                        5.9.7                h93ee506_2    conda-forge
qtconsole                 4.5.5                      py_0    conda-forge
readline                  7.0               hcfe32e1_1001    conda-forge
rhash                     1.3.6             h1de35cc_1001    conda-forge
scipy                     1.3.1            py36h7e0e109_2    conda-forge
send2trash                1.5.0                      py_0    conda-forge
setuptools                41.6.0                   py36_1    conda-forge
sip                       4.19.8          py36h0a44026_1000    conda-forge
six                       1.13.0                   py36_0    conda-forge
sqlite                    3.30.1               ha441bb4_0  
swig                      3.0.12               h1cb45b9_3    anaconda
terminado                 0.8.2                    py36_0    conda-forge
testpath                  0.4.4                      py_0    conda-forge
tk                        8.6.9             h2573ce8_1003    conda-forge
tornado                   6.0.3            py36h0b31af3_0    conda-forge
traitlets                 4.3.3                    py36_0    conda-forge
wcwidth                   0.1.7                      py_1    conda-forge
webencodings              0.5.1                      py_1    conda-forge
wheel                     0.33.6                   py36_0    conda-forge
widgetsnbextension        3.5.1                    py36_0    conda-forge
xz                        5.2.4             h1de35cc_1001    conda-forge
zeromq                    4.3.2                h6de7cb9_2    conda-forge
zipp                      0.6.0                      py_0    conda-forge
zlib                      1.2.11            h0b31af3_1006    conda-forge

```


```
$ helics_app --version
```

**Additional context and information**
(e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

--- a/interfaces/matlab/matlab_maps.i
+++ b/interfaces/matlab/matlab_maps.i
@@ -252,7 +252,7 @@ static void throwHelicsMatlabError(helics_error *err) {
 
 // Set argument to NULL before any conversion occurs
 %typemap(check)(void *data, int maxDatalen, int *actualSize) {
-    $2=helicsSubscriptionGetValueSize(arg1)+2;
+    $2 = helicsInputGetRawValueSize(arg1) + 2;
     $1 =  malloc($2);
 }
 

--- a/interfaces/matlab/matlab_maps.i
+++ b/interfaces/matlab/matlab_maps.i
@@ -257,7 +257,7 @@ static void throwHelicsMatlabError(helics_error *err) {
 }
 
 %typemap(argout) (void *data, int maxDatalen, int *actualSize) {
- if (--resc>=0) *resv++ = SWIG_FromCharPtrAndSize($1,*$3);
+ if (--resc>=0) *resv++ = SWIG_FromCharPtrAndSize((char*)$1,*$3);
 }
 
 // typemap for raw message data functions
@@ -276,5 +276,5 @@ static void throwHelicsMatlabError(helics_error *err) {
 }
 
 %typemap(argout) (void *data, int maxMessagelen, int *actualSize) {
- if (--resc>=0) *resv++ = SWIG_FromCharPtrAndSize($1,*$3);
+ if (--resc>=0) *resv++ = SWIG_FromCharPtrAndSize((char*)$1,*$3);
 }

